src/pages/Cart/Cart.jsx

```diff
@@ -1,10 +1,76 @@
-import { useAuth } from "../../contexts";
+import "./Cart.css";
+import { useAuth, useCart } from "../../contexts";
+import { HorizontalCard, Button } from "../../components";
 export const Cart = () => {
   const { checkUserLogin } = useAuth();
   checkUserLogin();
+  const { cartState } = useCart();
+  const { cart } = cartState;
+
+  const totalItems = cart.reduce((acc, curr) => {
+    acc = acc + curr.qty;
+    return acc;
+  }, 0);
+
+  const totalCostOfProduct = cart.reduce((acc, curr) => {
+    acc = acc + curr.price * curr.qty;
+    return acc;
+  }, 0);
+
+  const totalDiscount = cart.reduce((acc, curr) => {
+    acc = acc + curr.discount * curr.qty;
+    return acc;
+  }, 0);
```

Comment:
instead of multiple iteration of same array, use reduce in one time and calculate everything.

Reply:
Will Implement That